File: athena-core/src/models/spec/validations/enum_field.rs
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize, Debug, Default, PartialEq, Clone)]
pub struct EnumField {
    validations: Option<EnumValidation>,
}

#[derive(Serialize, Deserialize, Debug, PartialEq, Default, Clone)]
pub struct EnumValidation {
    enum_name: String,
    enum_localization: EnumLocalization,
}

#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
pub enum EnumLocalization {
    #[serde(rename = "method")]
    Method,
    #[serde(rename = "service")]
    Service,
    #[serde(rename = "global")]
    Global,
}

impl Default for EnumLocalization {
    fn default() -> Self {
        EnumLocalization::Global
    }
}

File: athena-gui/src/models/methodFields/enum_field.ts
import { z } from 'zod';
import { APISpecFieldTypeKeysEnum, BaseAPISpecMethodField } from './baseField';

export const EnumLocalization = z.enum(['method', 'service', 'global']);
export type EnumLocalization = z.infer<typeof EnumLocalization>;

export const EnumField = BaseAPISpecMethodField.extend({
    field_type: z.literal(APISpecFieldTypeKeysEnum.enum.Enum),
    validations: z.object({
        enum_name: z.string(),
        enum_localization: EnumLocalization.default('global'),
    })
});
export type EnumField = z.infer<typeof EnumField>;

